src/findings_data_import.py

```diff
@@ -49,12 +53,372 @@ def move_processed_file(s3_client, bucket, folder, filename):
+        if (
+            "RVA ID" not in finding.keys()
+            or "NCATS ID" not in finding.keys()
+            or "Severity" not in finding.keys()
+        ):
+            logging.warning(
+                'Skipping record %d. Missing "RVA ID", "NCATS ID", or "Severity" field.',
+                index,
+            )
+            continue
+
+        # Get RVA ID in format DDDD([.-]D+) from the end of the "RVA ID" field.
+        rvaId = re.search(r"(\d{4})(?:[.-](\d+))?$", finding["RVA ID"])
+        if rvaId:
+            rID = f"RV{rvaId.group(1)}"
+            if rvaId.group(2) is not None:
+                rID += f".{rvaId.group(2)}"
+            finding["RVA ID"] = rID
+        else:
+            logging.warning(
+                'Skipping record %d: Unable to extract valid RVA ID from "%s"',
+                index,
+                finding["RVA ID"],
+            )
+            continue
+        # flag this as V1 so update knows how to handle it, and its clear to folks viewing the data downstream
+        finding["Schema"] = V1_SCHEMA
+        valid_findings.append(finding)
+
+    return valid_findings
+
+
+def validate_v2_findings(findings_data: dict):
+    """Validate a list of V1 findings, discarding invalid entries (such as those with no severity which are not explicitly 'findings').
+
+    Parameters
+    ----------
+    findings_data : dict
+        The findings dictionary pulled from a findings JSON file (v2-style)
+
+    field_map_dict: dict
+        The dictionary of replacement rules for field names in findings_data (used only in V1 style processing)
+
+    Returns
+    -------
+    list : A list of findings objects from findings_data that pass validation
+    """
+    try:
+        if not findings_data or not isinstance(findings_data, dict):
+            logging.warning("Received an empty or invalid finding object, skipping.")
+            raise ValueError("Received an empty or invalid object.")
+
+        # work with v1 and v2. If has NCATS ID OR findings the document is probably OK
+        if "id" not in findings_data.keys() or "findings" not in findings_data.keys():
+            logging.warning('Skipping record. Missing "id" or "findings" field.')
+            raise ValueError("Missing id or findings field in v2 findings object")
+
+        # Get RVA ID in format DDDD([.-]D+) from the end of the "RVA ID" field.
+        rvaId = re.search(r"(\d{4})(?:[.-](\d+))?$", findings_data["id"])
+        if rvaId:
+            rID = f"RV{rvaId.group(1)}"
+            if rvaId.group(2) is not None:
+                rID += f".{rvaId.group(2)}"
+            findings_data["id"] = rID
+        else:
+            logging.warning(
+                "Skipping record. Unable to extract valid RVA ID from %s",
+                findings_data["id"],
+            )
+            raise ValueError("Unable to parse RVA id from finding object")
+        findings_data["Schema"] = V2_SCHEMA
```

Comment:
We're not gonna change this at this time. If it is an issue I suggest we do it as part of a larger overhaul that covers how V1 and V2 *should* work, or after usage for a while we just rip out any "V1" style operations at all since this schema should be phasing out.

Reply:
Only the person who originally created the conversation (or someone that person directed) should be resolving conversations.